sdk/python/feast/feature_store.py

```diff
@@ -541,6 +544,23 @@ def get_historical_features(
         all_on_demand_feature_views = self._registry.list_on_demand_feature_views(
             project=self.project
         )
+        referenced_odfvs = [
+            x for x in all_on_demand_feature_views if x.name in _feature_refs
+        ]
+        # Support validation of SQL
+        if type(entity_df) == pd.DataFrame:
+            entity_pd_df = cast(pd.DataFrame, entity_df)
+            for odfv in referenced_odfvs:
+                odfv_inputs = odfv.inputs.values()
+                for odfv_input in odfv_inputs:
+                    if type(odfv_input) == RequestDataSource:
+                        request_data_source = cast(RequestDataSource, odfv_input)
+                        for feature_name in request_data_source.schema.keys():
+                            if feature_name not in entity_pd_df.columns:
+                                raise RequestDataNotFoundInEntityDfException(
+                                    feature_name=feature_name,
+                                    feature_view_name=odfv.name,
+                                )
```

Comment:
Can we add tests to assert this failure?

Reply:
Done